openassessment/xblock/validation.py

```diff
@@ -48,8 +48,13 @@ def _duplicates(items):
 def _is_valid_assessment_sequence(assessments):
     """
     Check whether the sequence of assessments is valid.
-    For example, we currently allow self-assessment after peer-assessment,
-    but do not allow peer-assessment before self-assessment.
+    TODO: this documentation is horribly outdated, what defines a valid sequence anymore?
```

Comment:
I'm particularly interested what people think here - I feel that explicitly hard-coding all the possible sequences is already excessively unwieldy, are the rules I'm working with valid? Should there be any other validation assumptions?